src/Microsoft.ServiceFabric.Client.Http/ImageStoreClient_.cs

```diff
@@ -37,6 +61,122 @@ internal partial class ImageStoreClient
                 cancellationToken);
         }
 
+        /// <inheritdoc />
+        public Task UploadFileChunkAsync(
+            byte[] fileChunkToUpload,
+            string pathInImageStore,
+            Guid? sessionId,
+            long startBytePosition,
+            long endBytePosition,
+            long length,
+            long? serverTimeout = 60,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return this.UploadFileChunkAsync(
+                fileChunkToUpload,
+                pathInImageStore,
+                sessionId,
+                endBytePosition,
+                endBytePosition,
+                length,
+                null,
+                serverTimeout,
+                cancellationToken);
+        }
+
+        /// <inheritdoc />
+        public async Task UploadApplicationPackageAsync(
+            string applicationPackagePath,
+            bool compressPackage = false,
+            string applicationPackagePathInImageStore = default(string),
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            applicationPackagePath.ThrowIfNull(nameof(applicationPackagePath));
+
+            if (!Directory.Exists(applicationPackagePath))
+            {
+                throw new InvalidOperationException($"Application package path {applicationPackagePath} not found.");
+            }
+
+            var absPkgPath = FileUtilities.GetAbsolutePath(applicationPackagePath);
+
+            if (compressPackage)
```

Comment:
Just noticed that your PR#10 has introduced a bug here, it's compressing the code, config, data packages in place, leaving uncompressed files there as well, which will upload everything. Could you please revert PR #10 or fix it properly with accepting a target dir where user wants to compress the package into or only uploading compressed files or compress to a temp folder.  This would need little more thought, in the meantime we can revert PR#10 and bring it back in a few PR.